src/common/cert_vfy.c

```diff
@@ -143,20 +143,29 @@ static X509_CRL *download_crl(const char *uri)
 static int verify_crl(X509_CRL * crl, X509_STORE_CTX * ctx)
 {
   int rv;
-  X509_OBJECT obj;
   EVP_PKEY *pkey = NULL;
   X509 *issuer_cert;
 
   /* get issuer certificate */
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
+  X509_OBJECT obj;
   rv = X509_STORE_get_by_subject(ctx, X509_LU_X509, X509_CRL_get_issuer(crl), &obj);
+#else
+  X509_OBJECT *obj = X509_OBJECT_new();
+  rv = X509_STORE_get_by_subject(ctx, X509_LU_X509, X509_CRL_get_issuer(crl), obj);
+#endif
   if (rv <= 0) {
     set_error("getting the certificate of the crl-issuer failed");
     return -1;
   }
   /* extract public key and verify signature */
   issuer_cert = X509_OBJECT_get0_X509((&obj));
```

Comment:
`X509_OBJECT_get0_X509()` seems to be introduced in 1.1.0 and not available in 1.0.2. Perhaps this could work.

```
#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
  issuer_cert = obj.data.x509
#else
  issuer_cert = X509_OBJECT_get0_X509((&obj));
#endif
```

Reply:
X509_OBJECT_get0_X509() is defined in https://github.com/OpenSC/pam_pkcs11/blob/master/src/common/pam-pkcs11-ossl-compat.h#L97 for OpenSSL < 1.1.0